chia/server/server.py

```diff
@@ -178,6 +179,9 @@ def my_id(self) -> bytes32:
     def set_received_message_callback(self, callback: Callable):
         self.received_message_callback = callback
 
+    def set_capabilities(self, capabilities: List[Tuple[uint16, str]]):
```

Comment:
Another documentation suggestion - for the input type of `capabilities`, would you consider something like:

```
class CapabilityParam:
    capability: Capability
    xyz_str: str
```

or some other structured thing, e.g. `CapabilityParam(capability: uint16, xyz_str)`

I understand preferring the latter, since there's less argument conversion up-front.


Also, I'm not sure just by looking what the string is for - is that a value passed on the wire to signify the protocol version for this set of capabilities?

Reply:
1. i don't mind doing a class for that
2. mariano did the capabilities code way back and i think the purpose was that we could version a type of capability, this pr and the new weight_proof will be the first to use the capabilities and i didn't find a use for the string

Reply:
The string is not used yet, but it could be used to say specific things about a capability, for example NONE_RESPONSE version, or partial support for a capability, etc. The only one we used so far was BASE with a value of "1". We can use this string however we see fit, but it's probably good to come up with a scheme for how to use it.